test/libgit2.jl

```diff
@@ -1375,11 +1377,17 @@ mktempdir() do dir
         @test LibGit2.checkused!(creds)
         @test creds.user == creds_user
         @test creds.pass == creds_pass
+        creds2 = LibGit2.UserPasswordCredentials(creds_user, creds_pass)
+        @test creds == creds2
+        LibGit2.reset!(creds)
+        @test !LibGit2.checkused!(creds)
```

Comment:
I think having the `reset!` and `checkused!` moved up to above. Do you want this here to ensure that equality works when the count fields don't match?